instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -119,25 +119,27 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
                 {"params", "path", "verb"},
             )
 
-            region_name = deep_getattr(instance, "meta.region_name")
-
-            meta = {
-                "aws.agent": "botocore",
-                "aws.operation": operation,
-                "aws.region": region_name,
-            }
-            for key, value in meta.items():
-                span.set_attribute(key, value)
-
-            result = original_func(*args, **kwargs)
-
-            span.set_attribute(
-                "http.status_code",
-                result["ResponseMetadata"]["HTTPStatusCode"],
-            )
-            span.set_attribute(
-                "retry_attempts", result["ResponseMetadata"]["RetryAttempts"],
-            )
+            if span.is_recording():
```

Comment:
Shouldn't is_recording also prevent the setting of Resource attributes near line 100?